src/promptflow/promptflow/_proxy/_proxy_factory.py

```diff
@@ -22,43 +22,4 @@ def __init__(self):
         pass
 
     def get_executor_proxy_cls(self, language: str) -> Type[AbstractExecutorProxy]:
-        return self.executor_proxy_classes[language]
-
-    @classmethod
-    def register_executor(cls, language: str, executor_proxy_cls: Type[AbstractExecutorProxy]):
-        """Register a executor proxy class for a specific program language.
-
-        This method allows users to register a executor proxy class for a particular
-        programming language. The executor proxy class will be used when creating an instance
-        of the BatchEngine for flows written in the specified language.
-
-        :param language: The flow program language of the executor proxy,
-        :type language: str
-        :param executor_proxy_cls: The executor proxy class to be registered.
-        :type executor_proxy_cls:  ~promptflow.batch.AbstractExecutorProxy
-        """
-        cls.executor_proxy_classes[language] = executor_proxy_cls
-
-    def create_executor_proxy(
-        self, flow_file, working_dir, connections, storage, language: str, **kwargs
-    ) -> AbstractExecutorProxy:
-        executor_proxy_cls = self.get_executor_proxy_cls(language)
-        return async_run_allowing_running_loop(
-            executor_proxy_cls.create,
-            flow_file,
-            working_dir,
-            connections=connections,
-            storage=storage,
-            **kwargs,
-        )
-
-    def create_inspector_proxy(self, language: str, **kwargs) -> AbstractInspectorProxy:
-        if language == FlowLanguage.Python:
-            from promptflow._proxy._python_inspector_proxy import PythonInspectorProxy
-
-            return PythonInspectorProxy()
-        elif language == FlowLanguage.CSharp:
-            from promptflow._proxy._csharp_inspector_proxy import CSharpInspectorProxy
-
-            return CSharpInspectorProxy()
-        raise ValueError(f"Unsupported language: {language}")
+        return self.executor_proxy_classes[language] if language is not None else ChatGroupOrchestratorProxy
```

Comment:
maybe we can change the param name `language` to `type` and then you can register `ChatGroupOrchestratorProxy` to type `chat`

Reply:
Updated.